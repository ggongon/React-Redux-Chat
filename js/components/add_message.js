import React, {PropTypes} from 'react';
import {connect} from 'react-redux';
import * as actions from '../actions';

export class AddMessage extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      messageText: ""
    }
    this.onChange = this.onChange.bind(this);
  }

  onKeyPress(e) {
    if (e.key === "Enter") {
      this.onSubmit(e);
    }
  }

  onChange(e) {
    this.setState({messageText: e.target.value});
  }

  onSubmit(e) {
    e.preventDefault();
    this.props.spy && this.props.spy();
    this.props.addMessage(this.props.currentRoomId, {
      name: this.props.username,
      message: this.state.messageText
    });
    this.setState({messageText: ''});

  }
  render() {
    return (
      <footer className="footer">
        <form className="flex row" onSubmit={this.onSubmit.bind(this)}>
          <input className="stretchy" type="text"
            placeholder="Type a message and hit enter..."
            value={this.state.messageText}
            onChange={this.onChange}
            onKeyPress={this.onKeyPress.bind(this)}
          />
          <button className="submit" type="submit">Add</button>
        </form>
      </footer>
    );
  }
}

const mapStateToProps = ({currentRoomId, username}) => {
  return {currentRoomId, username};
}
export default connect(mapStateToProps, actions)(AddMessage);
