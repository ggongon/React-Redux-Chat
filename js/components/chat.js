import React, {PropTypes} from 'react';
import RoomList from './room_list'
import Room from './room';

export default (props) => {
    return (
      <div id="app" className="flex row">
        <div id="left">
          <RoomList />
        </div>
        <div id="right">
          <Room />
        </div>
      </div>
    );
}
