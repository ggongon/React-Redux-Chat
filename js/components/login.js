import React, {Component} from 'react';
import {connect} from 'react-redux';
import * as actions from '../actions';


export class Login extends Component {

  loginUser(event) {
    event.preventDefault();
    // update the state.username
    this.props.authenticate(this.refs.login.value);
  }

  test() {
    this.props.spy && this.props.spy();
  }

  componentWillUpdate(nextProps) {
    if(nextProps.username) {
      this.context.router.history.push("/chat");
    }
  }

  render() {
    return (
      <div id="loginContainer" className="flex absCenter">
        <form  className="loginContent flex absCenter col" onSubmit={this.loginUser.bind(this)}>
          <input type="text" ref="login" name="login" placeholder="Username"/>
          {/* <input type="password" ref="password" name="password" placeholder="Password"/> */}
          <button type="submit" name="submit" onClick={this.test.bind(this)}>Chat now!</button>
        </form>
      </div>
    );
  }
}

Login.contextTypes = {
  router: React.PropTypes.object
}

function mapStateToProps({username}) {
  return {username}
}

export default connect(mapStateToProps, actions)(Login);
