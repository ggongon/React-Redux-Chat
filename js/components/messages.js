import React, {Component} from 'react';
import {connect} from 'react-redux';
import {fetchMessages} from '../actions'

export class Messages extends Component {
  constructor(props) {
    super(props);
  }

  componentWillMount () {
    if (this.props.currentRoomId === null) {
      // console.log("cwm:", this.props.currentRoomId)
      this.props.fetchMessages(this.props.currentRoomId);
    }
  }

  componentWillUpdate(nextProps) {
    // console.log("cwu:", nextProps.currentRoomId)
    if (nextProps.currentRoomId !== this.props.currentRoomId) {
      this.props.fetchMessages(nextProps.currentRoomId);
    }
  }

  renderMessages() {
    if (!this.props.messages) return;
    console.log(this.props)
    return this.props.messages.map((message, index) => {
      return (
        <li key={index} className={(this.props.username === message.name)?"txtR":""}>
          <div className="message">{message.message}</div>
          {message.name!==this.props.username && <div className="user">{message.name}</div>}
        </li>
      )
    })
  }

  render() {
    // console.log(this.props.currentRoomId, this.props.messages)
    if (!this.props.messages.length) return <div className="stretchy">No messages.</div>

    return (
      <div id="messages" className="stretchy">
        <ul>
          {this.renderMessages()}
        </ul>
      </div>
    );
  }
}

const mapStateToProps = ({username, currentRoomId, messages}) => {
  return {
    username, currentRoomId, messages
  }
}

export default connect(mapStateToProps, {fetchMessages})(Messages);
