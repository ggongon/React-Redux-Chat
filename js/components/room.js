import React, {PropTypes} from 'react';
import {connect} from 'react-redux';

import RoomInfo from './room_info';
import Messages from './messages';
import AddMessage from './add_message';


export class Room extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div className="flex col">
        <RoomInfo currentRoom={this.props.currentRoom}/>
        <Messages />
        <AddMessage />
      </div>
    );
  }
}

function mapStateToProps({currentRoom}) {
  return {currentRoom}
}
export default connect(mapStateToProps)(Room);
