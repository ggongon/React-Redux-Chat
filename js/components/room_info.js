import React, {PropTypes} from 'react';

/**
 * sample response:
 * const currentRoom =  {
     name: 'Tea Chats',
     id: 0,
     users: [ 'Ryan', 'Nick' ]
   }
 */

const RoomInfo = ({currentRoom}) => {

  if (!Object.keys(currentRoom).length) return <div>get a room</div>
  let users = currentRoom.users.join(", ");
  return (
    <div className="roomInfo center">
      <h2>{currentRoom.name}</h2>
      <h5>{users}</h5>
    </div>
  );
}

// RoomInfo.propTypes = {
//   currentRoom: PropTypes.object,
// }


export default RoomInfo;
