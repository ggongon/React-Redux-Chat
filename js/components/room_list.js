import React, {PropTypes} from 'react';
import {connect} from 'react-redux';
import * as actions from '../actions';

export class RoomList extends React.Component {

  constructor(props) {
    super(props);
    this.renderRooms = this.renderRooms.bind(this);
    this.onRoomClicked = this.onRoomClicked.bind(this);
  }

  componentWillMount() {
    this.props.fetchRooms();
  }

  onRoomClicked(roomId) {
    this.props.spy && this.props.spy();
    if (roomId !== this.props.currentRoomId) {
      this.props.setCurrentRoomId(roomId);
      this.props.fetchRoom(roomId);
    }
  }

  renderRooms() {
    if (this.props.rooms) {
      return this.props.rooms.map((room) => {
        return (
          <li key={room.id} onClick={() => this.onRoomClicked(room.id) }>
            {room.name}
          </li>
        )
      })
    }
  }

  render() {
    return (
      <div>
        <ul>
          {this.renderRooms(this.props.rooms)}
        </ul>
      </div>
    )
  }
}

// RoomList.PropTypes = {
//   rooms: PropTypes.array,
// }

function mapStateToProps({rooms, currentRoomId}) {
  return { rooms, currentRoomId }
}

export default connect(mapStateToProps, actions)(RoomList);
