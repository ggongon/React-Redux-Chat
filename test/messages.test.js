import React from 'react';
import renderer from 'react-test-renderer';
import {Messages} from '../js/components/Messages';
import {shallow} from 'enzyme';

const currentRoom =  {
  name: 'Tea Chats',
  id: 0,
  users: [ 'Ryan', 'Nick' ]
}

const mock_messages = [
  {
    "name": "Ryan",
    "message": "ayyyyy"
  },
  {
    "name": "Nick",
    "message": "lmao"
  }
]

const props = {
  currentRoom,
  messages: mock_messages,
  fetchMessages: () => mock_messages
};

test("should render correctly", () => {

  const messages = renderer.create(<Messages {...props} />);
  expect(messages).toMatchSnapshot();
})

test("should render all messages for the selected room", () => {
  const messages = shallow( <Messages {...props} /> );

  expect(messages.find("li").length).toEqual(mock_messages.length);

})

test('should render a section where user can type their own messages', () => {

});

it('should dispatch an action when user enters a message', () => {

});
