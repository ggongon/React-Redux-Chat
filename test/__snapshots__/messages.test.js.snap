// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className="stretchy"
  id="messages"
>
  <ul>
    <li>
      <div
        className="message"
      >
        ayyyyy
      </div>
      <div
        className="user"
      >
        Ryan
      </div>
    </li>
    <li>
      <div
        className="message"
      >
        lmao
      </div>
      <div
        className="user"
      >
        Nick
      </div>
    </li>
  </ul>
</div>
`;
